storybookjs/storybook
# Addon-Docs: Upgrade to MDX3

Closes #25182 





## What I did

- Remove the jest-transform-mdx import used for storyshots
- Remove support for CSF in MDX
- Remove usage of @storybook/mdx1-csf and @storybook/mdx2-csf
- Bring the compiler code from @storybook/mdx2-csf into addon docs, alongside with @mdx-js/mdx
- Use MDX 3
- Upgrade @storybook/docs-mdx to a version that uses MDX3

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing


1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Access docs stories, they should work
4. Build Storybook, it should work
5. Serve the built Storybook and check the stories, just in case

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release


This pull request has been released as version [`0.0.0-pr-25303-sha-7c4e18ec`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-25303-sha-7c4e18ec). Install it by pinning all your Storybook dependencies to that version.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-25303-sha-7c4e18ec`](https://npmjs.com/package/@storybook/cli/v/0.0.0-pr-25303-sha-7c4e18ec) |
| **Triggered by** | @yannbf |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`yann/upgrade-mdx`](https://github.com/storybookjs/storybook/tree/yann/upgrade-mdx) |
| **Commit** | [`7c4e18ec`](https://github.com/storybookjs/storybook/commit/7c4e18ec1cbc2b8bb58e93733dd42dc95eedcc92) |
| **Datetime** | Tue Jan  2 12:53:00 UTC 2024 (`1704199980`) |
| **Workflow run** | [7385987242](https://github.com/storybookjs/storybook/actions/runs/7385987242) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=25303`_
</details>
